diff -ur -x .git SpiceyPy/setup.py /Users/bogdan/git/SpiceyPy/setup.py
--- SpiceyPy/setup.py	2018-02-21 17:14:14.000000000 +0100
+++ /Users/bogdan/git/SpiceyPy/setup.py	2017-11-09 23:31:18.000000000 +0100
@@ -81,7 +81,6 @@
     """
 
     def run(self):
-        '''
         self.check_for_spice()
 
         print("Host OS: {0}".format(host_OS))
@@ -91,7 +90,7 @@
             self.windows_method()
         else:
             sys.exit("Unsupported OS: {0}".format(host_OS))
-        '''
+
         install.run(self)
 
     @staticmethod
