diff -ur -x .git SpiceyPy/setup.py /Users/bogdan/git/SpiceyPy/setup.py
--- SpiceyPy/setup.py	2019-02-25 07:47:06.000000000 +0100
+++ /Users/bogdan/git/SpiceyPy/setup.py	2019-02-25 07:38:26.000000000 +0100
@@ -60,7 +60,6 @@
 
     @staticmethod
     def get_cspice():
-        '''
         if InstallCSpice.check_for_spice():
             print("Host OS: {0}".format(host_OS))
             if is_unix:
@@ -69,7 +68,6 @@
                 InstallCSpice.windows_method()
             else:
                 sys.exit("Unsupported OS: {0}".format(host_OS))
-        '''
 
     @staticmethod
     def check_for_spice():
